go/tasks/pluginmachinery/k8s/config.go

```diff
@@ -0,0 +1,58 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	restclient "k8s.io/client-go/rest"
+)
+
+type ClusterConfig struct {
+	Name     string `json:"name" pflag:",Friendly name of the remote cluster"`
+	Endpoint string `json:"endpoint" pflag:", Remote K8s cluster endpoint"`
+	Auth     Auth   `json:"auth" pflag:"-, Auth setting for the cluster"`
+	Enabled  bool   `json:"enabled" pflag:", Boolean flag to enable or disable"`
+}
+
+type Auth struct {
+	Type      string `json:"type" pflag:", Authentication type"`
+	TokenPath string `json:"token-path" pflag:", Token path"`
+	CertPath  string `json:"cert-path" pflag:", Certificate path"`
```

Comment:
I know it's not consistent across the codebase, but I'm trying to make new configs at least be camelCase
```suggestion
	TokenPath string `json:"tokenPath" pflag:",Token path"`
	CACertPath  string `json:"caCertPath" pflag:",Certificate Authority path"`
```